### DISCUSSION TITLE
Vendure provider for Nextjs Commerce Storefront

### DISCUSSION THREAD

**Discussion by megataps** [vendurehq/vendure#852] [Q&A]
Hi @michaelbromley 

I have tried install Vendure and Nextjs Commerce with Vendeure Provider branch on local. But I got an issue:
`
Error: queryA ENOTFOUND localhost
    at QueryReqWrap.onresolve [as oncomplete] (dns.js:203:19) {
  errno: undefined,
  code: 'ENOTFOUND',
  syscall: 'queryA',
  hostname: 'localhost',
  url: 'http://localhost:3001/shop-api',
  opts: {
    method: 'POST',
    headers: Headers { [Symbol(map)]: [Object: null prototype] },
    body: '{"query":"\\n  query getAllProducts($input: SearchInput!) {\\n    search(input: $input) {\\n      items {\\n        ...SearchResult\\n      }\\n    }\\n  }\\n  \\n  fragment SearchResult on SearchResult {\\n    productId\\n    productName\\n    description\\n    description\\n    slug\\n    sku\\n    currencyCode\\n    productAsset {\\n      id\\n      preview\\n    }\\n    priceWithTax {\\n      ... on SinglePrice {\\n        value\\n      }\\n      ... on PriceRange {\\n        min\\n        max\\n      }\\n    }\\n  }\\n\\n","variables":{"input":{"take":12,"groupByProduct":true}}}',
    agent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 4,
      _maxListeners: undefined,
      defaultPort: 80,
      protocol: 'http:',
      options: [Object],
      requests: {},
      sockets: {},
      freeSockets: {},
      keepAliveMsecs: 1000,
      keepAlive: true,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      freeSocketKeepAliveTimeout: 15000,
      timeout: 30000,
      socketActiveTTL: null,
      createSocketCount: 0,
      createSocketCountLastCheck: 0,
      createSocketErrorCount: 0,
      createSocketErrorCountLastCheck: 0,
      closeSocketCount: 0,
      closeSocketCountLastCheck: 0,
      errorSocketCount: 0,
      errorSocketCountLastCheck: 0,
      requestCount: 0,
      requestCountLastCheck: 0,
      timeoutSocketCount: 0,
      timeoutSocketCountLastCheck: 0,
      [Symbol(kCapture)]: false
    },
    redirect: 'manual',
    onRedirect: [Function (anonymous)]
  }
}
`
<img width="877" alt="Screen Shot 2021-04-30 at 2 12 15 PM" src="https://user-images.githubusercontent.com/7032500/116661131-2dab1800-a9be-11eb-9b26-ab89dd93a3f2.png">
<img width="1047" alt="Screen Shot 2021-04-30 at 2 12 24 PM" src="https://user-images.githubusercontent.com/7032500/116661141-30a60880-a9be-11eb-98c3-cef588c10d63.png">

Do you know what is it?

**Comment by michaelbromley** [comment]
Did you follow the instructions here: https://github.com/vendure-ecommerce/commerce/tree/vendure-provider/framework/vendure#usage ?

**Reply by megataps** [reply]
Yes, I did. As you can see on the attached image above, the Commerce Storefront has been up, but It cannot load product list from Vendure shop api.

Here is the image from my terminal.
<img width="1047" alt="Screen Shot 2021-04-30 at 2 29 23 PM" src="https://user-images.githubusercontent.com/7032500/116662877-8aa7cd80-a9c0-11eb-84cf-c2c8a1808eb0.png">

**Reply by michaelbromley** [reply]
Is `http://localhost:3001/shop-api` accessible? Do you see the graphql playground if you put it in your browser address bar?

**Comment by megataps** [comment]
Of course the url is accessible. I have tested on Postman as attached image.

<img width="746" alt="Screen Shot 2021-04-30 at 8 31 19 PM" src="https://user-images.githubusercontent.com/7032500/116702263-15092500-a9f3-11eb-9324-18851342797e.png">

**Reply by michaelbromley** [reply]
Apologies if my questions seem too simplistic. I often run into help requests where the simple things haven't yet been checked into.

I did a search on the error and came up with a few results - check these out if you didnt already see them:

* https://github.com/vercel/next.js/discussions/14604
* https://blog.andrewmmc.com/node-js-error-getaddrinfo-enotfound-localhost-b7ee35e1bb60

**Reply by megataps** [reply]
Thanks @michaelbromley 

I have updated the `.env.local` like this:
`COMMERCE_PROVIDER=vendure`
`NEXT_PUBLIC_VENDURE_SHOP_API_URL=http://127.0.0.1:3001/shop-api`

and `next.config.js` like this:
`images: {
    domains: ['127.0.0.1'],
  },
 `
It worked.
